src/ExtendedTableWidget.cpp

```diff
@@ -28,60 +28,131 @@ ExtendedTableWidget::ExtendedTableWidget(QWidget* parent) :
 
 void ExtendedTableWidget::copy()
 {
-    // Get list of selected items
-    QItemSelectionModel* selection = selectionModel();
-    QModelIndexList indices = selection->selectedIndexes();
+    QModelIndexList indices = selectionModel()->selectedIndexes();
 
-    // Abort if there's nothing to copy
-    if(indices.size() == 0)
-    {
+    //Abort if there's nothing to copy
+    if (indices.isEmpty())
         return;
-    } else if(indices.size() == 1) {
-        qApp->clipboard()->setText(indices.front().data().toString());
+    qSort(indices);
+
+    // Check whether selection is rectangular
+    QItemSelection rect(indices.first(), indices.last());
+    bool correct = true;
+
+    foreach (const QModelIndex& index, indices)
+        if (!rect.contains(index)) {
+            correct = false;
+            break;
+        }
+
+    foreach (const QModelIndex& index, rect.indexes())
+        if (!indices.contains(index)) {
+            correct = false;
+            break;
+        }
+
+    if (!correct) {
+        QMessageBox::warning(this, QApplication::applicationName(),
+                             tr("This function cannot be used with multiple independent selections."));
         return;
     }
 
-    // Sort the items by row, then by column
-    qSort(indices);
+    m_buffer.clear();
+
+    // If any of the cells contain binary data - we use inner buffer
+    bool containsBinary = false;
+    SqliteTableModel* m = qobject_cast<SqliteTableModel*>(model());
+    foreach (const QModelIndex& index, indices)
+        if (m->isBinary(index)) {
+            containsBinary = true;
+            break;
+        }
+
+    if (containsBinary) {
+        qApp->clipboard()->clear();
+        // Copy selected data into inner buffer
+        int columns = indices.last().column() - indices.first().column() + 1;
+        while (!indices.isEmpty()) {
+            QByteArrayList lst;
```

Comment:
Never mind, it'll all be fixed.
It's not even ready.

I am astonished though, why my build didn't fail.